nano/lib/numbers.cpp

```diff
@@ -939,6 +955,16 @@ nano::public_key::operator nano::hash_or_account const & () const
 	return reinterpret_cast<nano::hash_or_account const &> (*this);
 }
 
+bool nano::public_key::operator== (std::nullptr_t) const
+{
+	return bytes == null ().bytes;
```

Comment:
If we have a static object for the null account then this comparison can become a simple pointer comparison and there will be no temporary object involved. It is more efficient and it removes the theoretical possibility of a value clash.

Reply:
I don't believe there's a huge penalty being paid for this comparison, but since you shouldn't even be able to create the null account in a different way than using `nano::account::null ()` (or directly `nano::hardened_constants::get ().not_an_account` for that matter), then I guess it's very safe to use an address comparison -- therefore I addressed this as well.

Reply:
The difference is that this PR would affect only a 3-4 files whereas now it affects 36 files. But, I do not feel very strongly about it.